src/grammars/syntax.md

```diff
@@ -450,6 +450,48 @@ raw_string_interior = {
 
 ["raw string literals"]: https://doc.rust-lang.org/book/second-edition/appendix-02-operators.html#non-operator-symbols
 
+### Indentation-Sensitive Languages
+
+In conjunction with some extra helpers, the stack can be used to allow parsing indentation-sensitive languages, such as Python.
+
+The general idea is that you store the leading whitespace on the stack with `PUSH` and then use `PEEK_ALL` to match *all* of the whitespace on subsequent lines.
+
+When exiting an indented block, use `DROP` to remove the stack entry without needing to match it.
+
+An example grammar demonstrating this concept is given here:
+
+```pest
+Grammar = { SOI ~ NEWLINE* ~ BlockContent* ~ NEWLINE* ~ EOI }
+
+NewBlock = _{
+    // The first line in the block
+    PEEK_ALL ~ PUSH("  "+ | "\t"+) ~ BlockContent ~
+    // Subsequent lines in the block
+    (PEEK_ALL ~ BlockContent)* ~
+    // Remove the last layer of indentation from the stack when exiting the block
+    DROP
+}
+
+BlockName = { ASCII_ALPHA+ }
+
+BlockContent = {
+    BlockName ~ (NEWLINE | EOI) ~ NewBlock*
+}
+```
+
+This matches texts such as the following, whilst preserving indentation structure:
+
+```
+Hello
+  This
+    Is
+    An
+  Indentation
+    Sensitive
+      Language
+Demonstration
```

Comment:
This example may not a good case to showing up the stack use case.

With this case parse:

```pest
block = { identifier* }
identifier = @{ ASCII_ALPHA+ }
WHITESPACE = _{ " " | "\t" | NEWLINE }
```

Reply:
I think it tries to show how to parse the structure?
```
- Grammar
  - BlockContent
    - BlockName: "Hello"
    - BlockContent
      - BlockName: "This"
      - BlockContent > BlockName: "Is"
      - BlockContent > BlockName: "An"
    - BlockContent
      - BlockName: "Indentation"
      - BlockContent
        - BlockName: "Sensitive"
        - BlockContent > BlockName: "Language"
  - BlockContent
    - BlockName: "Demonstration"
    - EOI: ""
  - EOI: ""
```

Reply:
Yes, it's intended to preserve the structure (hence indentation-**sensitive** languages). I realise this will still leave quite a bit of work on the table for the reader, but at least it will be possible without having to find the extra tools mentioned in some issue on GitHub somewhere. (At first, I thought parsing indentation-sensitive languages was impossible with Pest, but eventually I found a comment that off-hand mentioned that these tools had been added for this purpose, but the docs didn't have a mention)